### ISSUE TITLE
ARC :: signer died because of a message

### ISSUE THREAD

**Issue by Philippe34** [sympa-community/sympa#988]
ARC features is on in my sympa.conf since several months. But a few days ago, a message could make an issue in sympa which could not send any messages anymore.

Version
-------
6.2.52

Installation method
-------------------
It is a centOS 7.8.2003 and I installed sympa with rpm repository

Expected behavior
-----------------
Yesterday, an user need help because the subscribers does not receive any messages since last week.
I did not know, sympa processes were always running for me.
I could notice that in Aug 27 09:37, a message sent to a list made an issue in Mail::AuthenticationResults::Parser and the consequences were that sympa didn't deliver any messages since a week. 
I was very annoyied for the users who could lost important messages.

Actual behavior
---------------
How did I solve this issue in the instant ?

I disabled arc feature in sympa.conf
arc_feature    off

Then I restarted sympa and immediatly all the suffering messages were delivered.
I reenabled arc_feature in sympa.conf, but, I disabled this feature to the list at the origin of the problem

Do you this is enough for this incident did not happen again in the futur ?

Is there a possibility to prevent this issue in sympa code ?

Additional information
----------------------
From the archives in sympa, if you want, I can provide the message which made the issue.

Sympa logs
``` code
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: notice Sympa::Spindle::ProcessIncoming::_twist() Processing Sympa::Message <list.1598513851.7776>; envelope_sender=no-reply@bounce.slideshare.net; message_id=5f473dd44630b_64ad3f87d856b5f413698d7@bounce.slideshare.net; sender=no-reply@bounce.slideshare.net
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: notice Sympa::Spindle::ProcessIncoming::_twist() Processing Sympa::Message <list.1598513851.7776>; envelope_sender=no-reply@bounce.slideshare.net; message_id=5f473dd44630b_64ad3f87d856b5f413698d7@bounce.slideshare.net; sender=no-reply@bounce.slideshare.net
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: info Sympa::Spindle::DoMessage::_twist() Processing message Sympa::Message <list.1598513851.7776/shelved:arc_cv=none> for Sympa::List <list@domain.fr> with priority 5, <5f473dd44630b_64ad3f87d856b5f413698d7@bounce.slideshare.net>
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: info Sympa::Spindle::DoMessage::_twist() Processing message Sympa::Message <list.1598513851.7776/shelved:arc_cv=none> for Sympa::List <list@domain.fr> with priority 5, <5f473dd44630b_64ad3f87d856b5f413698d7@bounce.slideshare.net>
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: notice Sympa::Spool::store() Sympa::Message <list.1598513851.7776/shelved:arc_cv=none;dkim_sign> is stored into Sympa::Spool::Archive as <1598513851.1598513852.057307.list@domain.fr,2832,6607>
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: notice Sympa::Spool::store() Sympa::Message <list.1598513851.7776/shelved:arc_cv=none;dkim_sign> is stored into Sympa::Spool::Archive as <1598513851.1598513852.057307.list@domain.fr,2832,6607>
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: notice Sympa::Spool::Outgoing::store() Message Sympa::Message <list.1598513851.7776/shelved:arc_cv=none;dkim_sign;dmarc_protect> is stored into bulk spool as <5.5.1598513851.1598513852.098266.list@domain.fr_z,2832,2514>
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: notice Sympa::Spindle::ToList::_send_msg() No VERP subscribers left to distribute message to list Sympa::List <list@domain.fr>
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: notice Sympa::Spool::Outgoing::store() Message Sympa::Message <list.1598513851.7776/shelved:arc_cv=none;dkim_sign;dmarc_protect> is stored into bulk spool as <5.5.1598513851.1598513852.098266.list@domain.fr_z,2832,2514>
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: notice Sympa::Spindle::ToList::_send_msg() No VERP subscribers left to distribute message to list Sympa::List <list@domain.fr>
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: info Sympa::Spindle::ToList::_twist() Message Sympa::Message <list.1598513851.7776/shelved:arc_cv=none;dkim_sign> for Sympa::List <list@domain.fr> from no-reply@bounce.slideshare.net accepted (0.10 seconds, 1 sessions, 4 subscribers), message ID=5f473dd44630b_64ad3f87d856b5f413698d7@bounce.slideshare.net, size=26172
Aug 27 09:37:32 vm-listhost sympa_msg[2832]: info Sympa::Spindle::ToList::_twist() Message Sympa::Message <list.1598513851.7776/shelved:arc_cv=none;dkim_sign> for Sympa::List <list@domain.fr> from no-reply@bounce.slideshare.net accepted (0.10 seconds, 1 sessions, 4 subscribers), message ID=5f473dd44630b_64ad3f87d856b5f413698d7@bounce.slideshare.net, size=26172
Aug 27 09:37:32 vm-listhost archived[2953]: notice Sympa::Spindle::ProcessArchive::_twist() Processing Sympa::Message <1598513851.1598513852.057307.list@domain.fr,2832,6607/shelved:arc_cv=none;dkim_sign>; sender: no-reply@bounce.slideshare.net; message ID: 5f473dd44630b_64ad3f87d856b5f413698d7@bounce.slideshare.net
Aug 27 09:37:32 vm-listhost archived[2953]: notice Sympa::Spindle::ProcessArchive::_twist() Archiving Sympa::Message <1598513851.1598513852.057307.list@domain.fr,2832,6607/shelved:arc_cv=none;dkim_sign> for list Sympa::List <list@domain.fr>
Aug 27 09:37:32 vm-listhost archived[2953]: notice Sympa::Archive::store() Message Sympa::Message <1598513851.1598513852.057307.list@domain.fr,2832,6607/shelved:arc_cv=none;dkim_sign> is stored into archive Sympa::Archive <list@domain.fr/2020-08> as <7>
Aug 27 09:37:32 vm-listhost archived[2953]: notice Sympa::Spindle::ProcessArchive::_twist() Processing Sympa::Message <1598513851.1598513852.057307.list@domain.fr,2832,6607/shelved:arc_cv=none;dkim_sign>; sender: no-reply@bounce.slideshare.net; message ID: 5f473dd44630b_64ad3f87d856b5f413698d7@bounce.slideshare.net
Aug 27 09:37:32 vm-listhost archived[2953]: notice Sympa::Spindle::ProcessArchive::_twist() Archiving Sympa::Message <1598513851.1598513852.057307.list@domain.fr,2832,6607/shelved:arc_cv=none;dkim_sign> for list Sympa::List <list@domain.fr>
Aug 27 09:37:32 vm-listhost archived[2953]: notice Sympa::Archive::store() Message Sympa::Message <1598513851.1598513852.057307.list@domain.fr,2832,6607/shelved:arc_cv=none;dkim_sign> is stored into archive Sympa::Archive <list@domain.fr/2020-08> as <7>
Aug 27 09:37:33 vm-listhost bulk[2959]: err main::#160 > Sympa::Spindle::spin#83 > Sympa::Spindle::ProcessOutgoing::_twist#398 > Sympa::Message::arc_seal#613 > Mail::DKIM::MessageParser::PRINT#54 > Mail::DKIM::ARC::Signer::finish_header#254 > Mail::AuthenticationResults::Parser::parse#42 > Mail::AuthenticationResults::Parser::_parse_authservid#150 > Carp::croak#100 DIED: unexpected token at /usr/share/perl5/vendor_perl/Mail/DKIM/ARC/Signer.pm line 254.
Aug 27 09:37:33 vm-listhost bulk[2959]: err main::#160 > Sympa::Spindle::spin#83 > Sympa::Spindle::ProcessOutgoing::_twist#398 > Sympa::Message::arc_seal#613 > Mail::DKIM::MessageParser::PRINT#54 > Mail::DKIM::ARC::Signer::finish_header#254 > Mail::AuthenticationResults::Parser::parse#42 > Mail::AuthenticationResults::Parser::_parse_authservid#150 > Carp::croak#100 DIED: unexpected token at /usr/share/perl5/vendor_perl/Mail/DKIM/ARC/Signer.pm line 254.
```

**Comment by ikedas** [comment]
Hi @Philippe34 ,

Could you please attach the message that caused death of bulk daemon?

And, if possible, please tell the versions of "Mail-DKIM" and "Mail-AuthenticationResults" Perl modules.

**Comment by Philippe34** [comment]
Hi @ikedas ,

Nice to see you again

I have the following versions:
perl-Mail-DKIM-0.55-0.20190413.RHEL7.noarch
perl-Mail-AuthenticationResults-1.20180923-2.el7.noarch

Here is the file in sympa/arc/ (without any changes)

[7.zip](https://github.com/sympa-community/sympa/files/5173747/7.zip)

**Comment by ikedas** [comment]
Hi @Philippe34 ,

Could you please apply [this patch](https://github.com/sympa-community/sympa/commit/fef332b.patch) and check if the problem will be solved?

**Comment by Philippe34** [comment]
Hi @ikedas ,

I've applied the patch to Message.pm
Like you, I found strange the Authentication-Results: with the reason "public key: not available"
So now in this situation, no ARC-Seal will be added.

I can check there is no disgression and ARC-Seal, ARC-Message-Signature and ARC-Authentication-Results are in the headers  when Message.pm is patched.
The issue was with slideshare.net. So I enabled arc_feature for the list and I can onlu wait for a new message coming from this domain.

It seems goods for me.

Thank you very much @ikedas, and when the sympa 6.2.58 will be released, i will upgrade to this version.
